catalog/confluence/catalog-confluence-source/src/main/java/org/codice/ddf/confluence/source/ConfluenceSource.java

```diff
@@ -0,0 +1,386 @@
+import java.io.InputStream;
+import java.io.Serializable;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+import org.boon.json.JsonFactory;
+import org.boon.json.ObjectMapper;
+import org.codice.ddf.configuration.PropertyResolver;
+import org.codice.ddf.confluence.api.SearchResource;
+import org.codice.ddf.cxf.SecureCxfClientFactory;
+import org.opengis.filter.sort.SortBy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.ContentType;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.ResultImpl;
+import ddf.catalog.filter.FilterAdapter;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.impl.SourceResponseImpl;
+import ddf.catalog.resource.ResourceNotFoundException;
+import ddf.catalog.resource.ResourceNotSupportedException;
+import ddf.catalog.resource.ResourceReader;
+import ddf.catalog.service.ConfiguredService;
+import ddf.catalog.source.FederatedSource;
+import ddf.catalog.source.SourceMonitor;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.util.impl.MaskableImpl;
+import ddf.security.encryption.EncryptionService;
+import ddf.security.permission.Permissions;
+
+public class ConfluenceSource extends MaskableImpl implements FederatedSource, ConfiguredService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfluenceSource.class);
+
+    private static final ObjectMapper MAPPER = JsonFactory.create();
+
+    private static final String USERNAME = "username";
+
+    private static final String PASSWORD = "password";
+
+    private static final String BASE_URL = "baseUrl";
+
+    private String endpointUrl;
+
+    private String configurationPid;
+
+    private String username;
+
+    private String password;
+
+    private String expandedSections = "";
+
+    private Boolean includeArchivedSpaces = false;
+
+    private Boolean includePageContent = false;
+
+    private Boolean excludeSpaces = false;
+
+    private List<String> confluenceSpaces = new ArrayList<>();
+
+    private final EncryptionService encryptionService;
+
+    private final FilterAdapter filterAdapter;
+
+    private final ResourceReader resourceReader;
+
+    private final ConfluenceInputTransformer transformer;
+
+    protected SecureCxfClientFactory<SearchResource> factory;
```

Comment:
Is this protected for testing?  Looks like you could just override `init` instead if that is the case.